src/app/codegen-data-model/EmberMetadata.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/codegen-data-model/EmberMetadata.h>
+
+#include <app/GlobalAttributes.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/endpoint-config-api.h>
+
+namespace chip {
+namespace app {
+namespace Ember {
+
+std::variant<const EmberAfCluster *,           // global attribute, data from a cluster
+             const EmberAfAttributeMetadata *, // a specific attribute stored by ember
+             CHIP_ERROR                        // error, this will NEVER be CHIP_NO_ERROR
+             >
+FindAttributeMetadata(const ConcreteAttributePath & aPath)
+{
+    for (auto & attr : GlobalAttributesNotInMetadata)
+    {
+
+        if (attr == aPath.mAttributeId)
```

Comment:
This was carried over  from FindAttributeMetadata from ember-compatibility-functions.cpp

https://github.com/project-chip/connectedhomeip/blob/master/src/app/util/ember-compatibility-functions.cpp#L256

My worry was that the logic may be intentional: we want to return cluster FIRST even if the attribute somehow magically exists, because global attributes need to be processed via the cluster. I am afraid that switching order may introduce sideffects that were not there before, so I prefer to keep the logic as is..

Reply:
> If we wanted to optimize more, we could have an IsGlobalAttribute() check on the id before we start doing the "not in metadata" check or something....

That is an easy compromise to avoid re-ordering and could be done in this PR.

Reply:
Added a IsGlobalAttribute check before the loop.